dask/dataframe/rolling.py

```diff
@@ -74,17 +93,44 @@ def map_overlap(func, df, before, after, *args, **kwargs):
     df_name = df._name
 
     dsk = df.dask.copy()
-    if before:
+
+    # Have to do the checks for too large windows in the time-delta case
+    # here instead of in `overlap_chunk`, since we can't rely on fix-frequency
+    # index
+
+    timedelta_partition_message = (
+        "Partition size is less than specified window. "
+        "Try using ``df.repartition`` to increase the partition size"
+    )
+
+    if before and isinstance(before, int):
         dsk.update({(name_a, i): (M.tail, (df_name, i), before)
                     for i in range(df.npartitions - 1)})
         prevs = [None] + [(name_a, i) for i in range(df.npartitions - 1)]
+    elif isinstance(before, datetime.timedelta):
+        # Assumes monotonic (increasing?) index
+        deltas = pd.Series(df.divisions).diff().iloc[1:-1]
+        if (before > deltas).any():
+            raise ValueError(timedelta_partition_message)
+        dsk.update({(name_a, i): (_tail_timedelta, (df_name, i), (df_name, i + 1), before)
+                    for i in range(df.npartitions - 1)})
+        prevs = [None] + [(name_a, i) for i in range(df.npartitions - 1)]
     else:
         prevs = [None] * df.npartitions
 
     if after:
         dsk.update({(name_b, i): (M.head, (df_name, i), after)
                     for i in range(1, df.npartitions)})
         nexts = [(name_b, i) for i in range(1, df.npartitions)] + [None]
+    elif isinstance(after, datetime.timedelta):
+        # TODO: Do we have a use-case for this? Pandas doesn't allow negative rolling windows
```

Comment:
If we include this though then we should test it.  It looks like things here are uncovered.